test/jdk/performance/client/RenderPerfTest/src/renderperf/RenderPerfTest.java

```diff
@@ -48,67 +61,186 @@
     private final static float HEIGHT = 800;
     private final static float R = 25;
     private final static int BW = 50;
     private final static int BH = 50;
+    private final static int IMAGE_W = (int) (WIDTH + BW);
+    private final static int IMAGE_H = (int) (HEIGHT + BH);
+
+    private final static String TEST_TEXT = TEXT_VERSION ? VERSION : "The quick brown fox jumps over the lazy dog";
+
     private final static int COUNT = 600;
-    private final static int CYCLE_DELAY = 3;
-    private final static int MAX_FRAME_CYCLES = 3000/CYCLE_DELAY;
+    private final static int MIN_COUNT = 20;
+    private final static int WARMUP_COUNT = MIN_COUNT;
+
+    private final static int DELAY = 1;
+    private final static int CYCLE_DELAY = DELAY;
+
+    private final static long MIN_MEASURE_TIME_NS = 1000L * 1000 * 1000;
+    private final static long MAX_MEASURE_TIME_NS = 6000L * 1000 * 1000;
+    private final static int MAX_FRAME_CYCLES = 3000 / CYCLE_DELAY;
 
     private final static int COLOR_TOLERANCE = 10;
-    private final static int MAX_MEASURE_CYCLES = 6000/CYCLE_DELAY;
 
-    private final static Color[] marker = {Color.RED, Color.BLUE, Color.GREEN, Color.YELLOW, Color.ORANGE, Color.MAGENTA};
+    private final static Color[] MARKER = {Color.RED, Color.BLUE, Color.GREEN};
+
+    private final static Toolkit TOOLKIT = Toolkit.getDefaultToolkit();
+
+    private final static long FRAME_MAX = 60;
+    private final static long FRAME_PREC_IN_NANOS = (1000L * 1000 * 1000) / (2L * FRAME_MAX);
 
     interface Configurable {
-        void configure(Graphics2D g2d);
+        void configure(Graphics2D g2d, boolean enabled);
     }
 
-    interface Renderable {
-        void setup(Graphics2D g2d);
-        void render(Graphics2D g2d);
-        void update();
+    final static class ConfigurableAA implements Configurable {
+        @Override
+        public void configure(final Graphics2D g2d, final boolean enabled) {
+            g2d.setRenderingHint(RenderingHints.KEY_ANTIALIASING,
+                    enabled ? RenderingHints.VALUE_ANTIALIAS_ON
+                            : RenderingHints.VALUE_ANTIALIAS_OFF);
+        }
+    }
+
+    final static class ConfigurableTextAA implements Configurable {
+        @Override
+        public void configure(final Graphics2D g2d, final boolean enabled) {
+            g2d.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING,
+                    enabled ? RenderingHints.VALUE_TEXT_ANTIALIAS_ON
+                            : RenderingHints.VALUE_TEXT_ANTIALIAS_OFF);
+        }
+    }
+
+    final static class ConfigurableTextLCD implements Configurable {
+        @Override
+        public void configure(final Graphics2D g2d, final boolean enabled) {
+            g2d.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING,
+                    enabled ? RenderingHints.VALUE_TEXT_ANTIALIAS_LCD_HRGB
+                            : RenderingHints.VALUE_TEXT_ANTIALIAS_DEFAULT);
+        }
```

Comment:
Although it makes no practical difference, I am wondering why in this case, enabled==false sets "DEFAULT" and in the case above it sets "OFF" ?

Reply:
Will check